src/UNL/UCBCN/Manager/Calendar.php

```diff
@@ -101,6 +110,32 @@ private function archiveEvents() {
                 $event->updateStatusWithCalendar($this->calendar, 'archived');
             }
         }
+
+        # check each past event to see if it is now current
+        foreach ($archived_events as $event) {
+            # we will consider it upcoming if any datetime is after today at midnight
+            $datetimes = $event->getDatetimes();
+            $move = FALSE;
+            foreach ($datetimes as $datetime) {
+                $recurring_dates = $datetime->getAllDates();
+                foreach($recurring_dates as $recurring_date) {
+                    if ($recurring_date->recurringdate >= date('Y-m-d')) {
+                        $move = TRUE;
+                        break;
+                    }
+                }
+
+                if ($datetime->starttime >= date('Y-m-d 00:00:00') || ($datetime->endtime != NULL && $datetime->endtime >= date('Y-m-d 00:00:00'))) {
+                    $move = TRUE;
+                    break;
+                }
+            }
+
+            # update the status with the calendar
+            if ($move) {
+                $event->updateStatusWithCalendar($this->calendar, 'posted');
```

Comment:
I'd still like to see a constant used for `'posted'`, rather than a magic variable.